orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -237,15 +251,18 @@ def backward(ctx, *grad_outputs):
                     backward_outputs = training_io_binding.get_outputs()
 
                     # Return input and initializer gradients
-                    num_user_input_grads = len(self._input_names_require_grad)
+                    num_user_input_grads = len(self._input_info.require_grad_names)
 
                     results = []
+                    require_grad_names_set = set(self._input_info.require_grad_names)
```

Comment:
Suggestion: Compute this set once when the `InputInfo` is built.

Reply:
I will create a PR that caches this set along with the initializers_names_to_train